core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -0,0 +1,230 @@
+        return new File(Jenkins.getActiveInstance().getRootDir(), "config.xml");
+    }
+
+    static File getLastExecVersionFile() {
+        return new File(Jenkins.getActiveInstance().getRootDir(), ".last_exec_version");
+    }
+
+    static File getInstallingPluginsFile() {
+        return new File(Jenkins.getActiveInstance().getRootDir(), ".installing_plugins");
+    }
+
+    private static String getCurrentExecVersion() {
+        if (Jenkins.VERSION.equals(Jenkins.UNCOMPUTED_VERSION)) {
+            // This should never happen!! Only adding this check in case someone moves the call to this method to the wrong place.
+            throw new IllegalStateException("Unexpected call to InstallUtil.getCurrentExecVersion(). Jenkins.VERSION has not been initialized. Call computeVersion() first.");
+        }
+        return Jenkins.VERSION;
+    }
+    
+    /**
+     * Returns a list of any plugins that are persisted in the installing list
+     */
+    @SuppressWarnings("unchecked")
+	public static synchronized @CheckForNull Map<String,String> getPersistedInstallStatus() {
+        File installingPluginsFile = getInstallingPluginsFile();
+        if(installingPluginsFile == null || !installingPluginsFile.exists()) {
+        	return null;
+        }
+        return (Map<String,String>)new XStream().fromXML(installingPluginsFile);
+    }
+    
+    /**
+     * Persists a list of installing plugins; this is used in the case Jenkins fails mid-installation and needs to be restarted
+     * @param installingPlugins
+     */
+    public static synchronized void persistInstallStatus(List<UpdateCenterJob> installingPlugins) {
+        File installingPluginsFile = getInstallingPluginsFile();
+    	if(installingPlugins == null || installingPlugins.isEmpty()) {
+    		installingPluginsFile.delete();
+    		return;
+    	}
+    	LOGGER.fine("Writing install state to: " + installingPluginsFile.getAbsolutePath());
+    	Map<String,String> statuses = new HashMap<String,String>();
+    	for(UpdateCenterJob j : installingPlugins) {
+    		if(j instanceof InstallationJob && j.getCorrelationId() != null) { // only include install jobs with a correlation id (directly selected)
+    			InstallationJob ij = (InstallationJob)j;
+    			InstallationStatus status = ij.status;
+    			String statusText = status.getType();
+    			if(status instanceof Installing) { // flag currently installing plugins as pending
+    				statusText = "Pending";
+    			}
+    			statuses.put(ij.plugin.name, statusText);
+    		}
+    	}
+        try {
+        	String installingPluginXml = new XStream().toXML(statuses);
+            FileUtils.write(installingPluginsFile, installingPluginXml);
+        } catch (IOException e) {
+            LOGGER.log(SEVERE, "Failed to save " + installingPluginsFile.getAbsolutePath(), e);
+        }
+    }
+
+    /**
+     * Call to remove any active install status
+     */
+	public static void clearInstallStatus() {
+		persistInstallStatus(null);
+	}
```

Comment:
@kzantow I think it would be good to update `InstallUtilTest` with some unit tests for the new "status" file utilities.